tests/eeid_plugin/enc/enc.c

```diff
@@ -0,0 +1,239 @@
+}
+
+static void _test_evidence_success(const oe_uuid_t* format_id)
+{
+    printf("====== running _test_evidence_success\n");
+
+    uint8_t* evidence = NULL;
+    size_t evidence_size = 0;
+    uint8_t* endorsements = NULL;
+    size_t endorsements_size = 0;
+    oe_claim_t* claims = NULL;
+    size_t claims_length = 0;
+
+    oe_result_t r = oe_get_evidence(
+        format_id,
+        OE_REPORT_FLAGS_REMOTE_ATTESTATION,
+        claims,
+        claims_length,
+        NULL,
+        0,
+        &evidence,
+        &evidence_size,
+        &endorsements,
+        &endorsements_size);
+
+#ifndef _OE_ENCLAVE_H
+    OE_UNUSED(_check_claims);
+    OE_TEST(r == OE_UNSUPPORTED);
+#else
+    OE_TEST(r == OE_OK);
+
+    // Verify evidence without endorsements.
+    OE_TEST(
+        oe_verify_evidence(
+            evidence,
+            evidence_size,
+            NULL,
+            0,
+            NULL,
+            0,
+            &claims,
+            &claims_length) == OE_OK);
+
+    OE_TEST(oe_free_claims_list(claims, claims_length) == OE_OK);
+
+    // Verify with endorsements.
+    OE_TEST(
+        oe_verify_evidence(
+            evidence,
+            evidence_size,
+            endorsements,
+            endorsements_size,
+            NULL,
+            0,
+            &claims,
+            &claims_length) == OE_OK);
+
+    OE_TEST(
+        host_verify(evidence, evidence_size, endorsements, endorsements_size) ==
+        OE_OK);
+
+    OE_TEST(oe_free_claims_list(claims, claims_length) == OE_OK);
+    OE_TEST(oe_free_endorsements(endorsements) == OE_OK);
+    OE_TEST(oe_free_evidence(evidence) == OE_OK);
+#endif
+}
+
+static void _test_get_evidence_fail()
+{
+    printf("====== running _test_get_evidence_fail\n");
+
+    uint8_t* evidence;
+    size_t evidence_size;
+
+    // Test get_evidence when plugin is unregistered.
+    oe_attester_t* attester = oe_eeid_plugin_attester();
+    OE_TEST(oe_unregister_attester(attester) == OE_OK);
+
+    OE_TEST(
+        oe_get_evidence(
+            &attester->base.format_id,
+            0,
+            NULL,
+            0,
+            NULL,
+            0,
+            &evidence,
+            &evidence_size,
+            NULL,
+            NULL) == OE_NOT_FOUND);
+
+    OE_TEST(oe_register_attester(attester, NULL, 0) == OE_OK);
+}
+
+static void _test_verify_evidence_fail()
+{
+    printf("====== running _test_verify_evidence_fail\n");
+
+    uint8_t* evidence;
+    size_t evidence_size;
+    uint8_t* endorsements;
+    size_t endorsements_size;
+    oe_claim_t* claims;
+    size_t claims_length;
+
+    oe_attester_t* attester = oe_eeid_plugin_attester();
+    OE_TEST(
+        oe_get_evidence(
+            &attester->base.format_id,
+            0,
+            NULL,
+            0,
+            NULL,
+            0,
+            &evidence,
+            &evidence_size,
+            &endorsements,
+            &endorsements_size) == OE_OK);
+
+    // Test verify_evidence with wrong sizes
```

Comment:
Sure, more tests are better, but it takes time to implement them. That particular one you describe is probably easy to add though.

Reply:
So I would strongly suggest adding whatever semantic tests are easy to add right now; some things are really easy to miss in implementation and code review, like whether the eeid data was actually being measured or not.

Reply:
Yeah, that's definitely missing. I'll have to write a bunch of infrastructure to get information out of the plugin data-structures, so this will take a bit of time.